packages/design-system-react/src/components/button-base/ButtonBase.tsx

```diff
@@ -78,6 +79,17 @@ export const ButtonBase = React.forwardRef<HTMLButtonElement, ButtonBaseProps>(
       return null;
     };
 
+    const renderContent = () => {
+      if (children && typeof children === 'string') {
+        return (
+          <Text color={TextColor.Inherit} asChild {...textProps}>
+            <span>{children}</span>
+          </Text>
```

Comment:
Here, we are ensuring that the child `Text` component inherits the button text color, renders as a `span` element, and accepts the `textProps` prop.